cristin-commons/src/main/java/no/unit/nva/cristin/common/handler/CristinQueryHandler.java

```diff
@@ -0,0 +1,102 @@
+package no.unit.nva.cristin.common.handler;
+
+import static no.unit.nva.cristin.common.model.Constants.DEFAULT_NUMBER_OF_RESULTS;
+import static no.unit.nva.cristin.common.model.Constants.DEFAULT_RESPONSE_MEDIA_TYPES;
+import static no.unit.nva.cristin.common.model.Constants.FIRST_PAGE;
+import static no.unit.nva.cristin.common.model.Constants.NUMBER_OF_RESULTS;
+import static no.unit.nva.cristin.common.model.Constants.PAGE;
+import static no.unit.nva.cristin.common.model.Constants.QUERY;
+import static nva.commons.core.attempt.Try.attempt;
+import com.google.common.net.MediaType;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import no.unit.nva.cristin.common.util.UriUtils;
+import nva.commons.apigateway.ApiGatewayHandler;
+import nva.commons.apigateway.RequestInfo;
+import nva.commons.apigateway.exceptions.BadRequestException;
+import nva.commons.core.Environment;
+import nva.commons.core.JacocoGenerated;
+
+@JacocoGenerated // TODO: When FetchCristinProjects extends this we will get testing for free
```

Comment:
It does not extend it yet, I try to make the PR as small as possible and refactoring FetchCristinProjects to use this will increase the size of the PR. If you want I can refactor it in this PR?

Reply:
Will be fixed in another PR